library/core/functions.general.php

```diff
@@ -1766,6 +1766,18 @@ function InSubArray($Needle, $Haystack) {
    }
 }
 
+if (!function_exists('IsHtml')) {
+   /**
+    * Determine whether text contains HTML.
+    *
+    * @param string $Text
+    * @return bool Whether HTML was detected.
+    */
+   function IsHtml($Text) {
+      return strpos($Text, '<') !== FALSE || (bool)preg_match('/&#?[a-z0-9]{1,10};/i', $Text);
```

Comment:
This is kind of a bad thing to make a global function as it really doesn't  check for HTML. The idea behind the check is for code that looks like html as an optimization. We don't need to filter code that doesn't have any tags.

Making this a global function is going to be misleading to other programmers that might want to use it for a more general purpose. Adding more functionality will kind of kill the entire reason for it in the first place as it will lose its optimization.